### PR TITLE
Hamburger Button 🍔

### PR DESCRIPTION
**Pull request by namniak**
(states + hooks)

### CODE REVIEW

**Comment by namniak** [comment]
image

File: templates/style/global.{{css}} [hunk 1]
```diff
@@ -14,6 +14,21 @@
...
+  font-size: 62.5%;
+}
+
+button {
+  border: none;
+  outline: none;
+  cursor: pointer;
+  background: transparent;
+  padding: 0;
+  font-family: akkurat, Helvetica, Arial, sans-serif;
```
Comment (iranreyes) [inline templates/style/global.{{css}}]: @namniak Do we included akkurat in the generator?

@njam3

File: templates/react/src/components/HamburgerButton/index.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.buttonState !== this.props.buttonState) {
+      this.setState({buttonState: nextProps.buttonState});
+    }
+    if (nextProps.isMouseOver !== this.props.isMouseOver) {
+      this.setState({isMouseOver: nextProps.isMouseOver});
+    }
+  }
+
+  _handleClick = (e) => {
```
Comment (iranreyes) [inline templates/react/src/components/HamburgerButton/index.js]: The other components in the generator don't use private methods with an underscore at the beginning; we should be consistent

Thoughts?

@namniak @njam3

File: templates/style/global.{{css}} (hunk 1 above)
Comment (neo) [reply templates/style/global.{{css}}]: nice catch, I think @namniak missed it.

File: templates/react/src/components/HamburgerButton/index.js (hunk 2 above)
Comment (neo) [reply templates/react/src/components/HamburgerButton/index.js]: agree without underscore

**Comment by namniak** [comment]
@iranreyes removed underscore and font
@njam3 refactored with JS animations.

**Pull request merged**